bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,1752 @@
+        ["T_el_0", "Electron temperature on axis", 2.196e+01, "keV", None, "Input"],
+        ["rho_ped_n", "Density pedestal r/a location", 9.4e-01, "dimensionless", None, "Input"],
+        ["rho_ped_t", "Temperature pedestal r/a location", 9.76e-01 , "dimensionless", None, "Input"],
+        ["n_el_ped", "Electron density pedestal height", 1.086e+20, "1/m^3", None, "Input"],
+        ["T_el_ped", "Electron temperature pedestal height", 3.74, "keV", None, "Input"],
+        ["alpha_n", "Density profile factor", 1.15, "dimensionless", None, "Input"],
+        ["alpha_t", "Temperature profile index", 1.905, "dimensionless", None, "Input"],
+        ["t_beta", "Temperature profile index beta", 2, "dimensionless", None, "Input"],
+        ["n_el_sep", "Electron density at separatrix", 1.5515e+19, "1/m^3", None, "Input"],
+        ["T_el_sep", "Electron temperature at separatrix", 4.8e-01, "keV", None, "Input"],
+        ["q_95", "Safety factor at 0.95 flux_surface", 4.9517, "dimensionless", None, "Input"],
+        ["rminor", "Minor_radius", 2.183, "m", None, "Input"],
+        ["kappa", "Elongation", 2.8, "dimensionless", None, "Input"],
+    ]
+    # fmt: on
+
+    def __init__(
+        self,
+        transport_solver: ChargedParticleSolver,
+        plasma_params: ParameterFrame,
+    ):
+        self.plasma_params = ParameterFrame(self.plasma_params)
+        self.plasma_params.update_kw_parameters(
+            plasma_params, f"{self.__class__.__name__} input"
+        )
+
+        self.transport_solver = transport_solver
+
+        self.collect_x_and_o_point_coordinates()
+
+        # Separatrix parameters
+        self.collect_separatrix_parameters()
+
+    def collect_x_and_o_point_coordinates(self):
+        """
+        Magnetic axis coordinates and x-point(s) coordinates.
+        """
+        o_point, x_point = self.transport_solver.eq.get_OX_points()
+        self.points = {
+            "x_point": {
+                "x": x_point[0][0],
+                "z_low": x_point[0][1],
+                "z_up": x_point[1][1],
+            },
+            "o_point": {"x": o_point[0][0], "z": round(o_point[0][1], 5)},
+        }
+        if self.points["x_point"]["z_low"] > self.points["x_point"]["z_up"]:
+            self.points["x_point"]["z_low"] = x_point[1][1]
+            self.points["x_point"]["z_up"] = x_point[0][1]
+
+    def collect_separatrix_parameters(self):
+        """
+        Some parameters related to the separatrix
+        """
+        self.separatrix = self.transport_solver.eq.get_separatrix()
+        # The two halves
+        self.sep_lfs = self.separatrix[0]
+        self.sep_hfs = self.separatrix[1]
+        # The mid-plane radii
+        self.x_sep_omp = self.transport_solver.x_sep_omp
+        self.x_sep_imp = self.transport_solver.x_sep_imp
+        # To move away from the mathematical separatrix which would
+        # give infinite connection length
+        self.r_sep_omp = self.x_sep_omp + SEP_CORRECTOR
```

Comment:
There is surely a better way to do this than via a CONSTANT...

Reply:
Having a connection length calculated correctly.
At the moment the calculation is very unstable near the separatrix. I need to move away from it by a random distance which varies according to the equilibrium.
An alternative would be a sort of iterator that moves further till the connetction length calculation "looks" correct. It doesn't seem worth, though.